packages/amqp/lib/AbstractAmqpExchangePublisher.ts

```diff
@@ -0,0 +1,50 @@
+import type * as Buffer from 'node:buffer'
+
+import { objectToBuffer } from '@message-queue-toolkit/core'
+import type { Options } from 'amqplib/properties'
+
+import type { AMQPPublisherOptions } from './AbstractAmqpPublisher'
+import { AbstractAmqpPublisher } from './AbstractAmqpPublisher'
+import type { AMQPDependencies } from './AbstractAmqpService'
+
+export type AMQPExchangePublisherOptions<MessagePayloadType extends object> = Omit<
+  AMQPPublisherOptions<MessagePayloadType>,
+  'creationConfig'
+> & {
+  exchange: string
+}
+
+export type AmqpExchangeMessageOptions = {
+  routingKey: string
+  publishOptions: Options.Publish
+}
+
+export class AbstractAmqpExchangePublisher<
+  MessagePayloadType extends object,
+> extends AbstractAmqpPublisher<MessagePayloadType, AmqpExchangeMessageOptions> {
+  constructor(
+    dependencies: AMQPDependencies,
+    options: AMQPExchangePublisherOptions<MessagePayloadType>,
+  ) {
+    super(dependencies, {
+      ...options,
+      // FixMe exchange publisher doesn't need queue at all
```

Comment:
will be addressed during auto-init rework, as we need to implement exchange and subscription support